### PR TITLE
cluster/dm: support deploying Prometheus witch custom command parameters

### PR DESCRIPTION
**Pull request by WalterWj**
### What problem does this PR solve? 
Some users want to turn on Prometheus's hot load feature. Or add some Prometheus commands manually

### What is changed and how it works?
img_v3_026g_107a4e8a-896d-4d8d-bf72-18df26c2131g
img_v3_026g_1a83581c-afcd-4488-934e-3f8ea6ad203g

### Check List 

Tests 

 - Unit test
 - Integration test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has exported function/method change
 - Has exported variable/fields change
 - Has interface methods change
 - Has persistent data change

Side effects

 - Possible performance regression
 - Increased code complexity
 - Breaking backward compatibility

Related changes

 - Need to cherry-pick to the release branch
 - Need to update the documentation

Release notes:

```release-note
NONE
```

### CODE REVIEW

File: pkg/cluster/spec/instance.go [hunk 1]
```diff
@@ -153,6 +153,7 @@ type BaseInstance struct {
 	Source     string
 	NumaNode   string
 	NumaCores  string
+	RunCommand string
```
Comment (srstack) [inline pkg/cluster/spec/instance.go:156]: should not be added to BaseInstance.

File: pkg/cluster/spec/monitoring.go [hunk 2]
```diff
@@ -60,6 +60,7 @@ type PrometheusSpec struct {
 	AdditionalScrapeConf  map[string]any         `yaml:"additional_scrape_conf,omitempty" validate:"additional_scrape_conf:ignore"`
 	ScrapeInterval        string                 `yaml:"scrape_interval,omitempty" validate:"scrape_interval:editable"`
 	ScrapeTimeout         string                 `yaml:"scrape_timeout,omitempty" validate:"scrape_timeout:editable"`
+	RunCommand            string                 `yaml:"run_command,omitempty" validate:"run_command:editable"`
```
Comment (srstack) [inline pkg/cluster/spec/monitoring.go:63]: Use string[] for better handling of multiple commands

**Comment by WalterWj** [comment]
image
image

File: pkg/cluster/spec/instance.go (hunk 1 above)
Comment (WalterWj) [reply pkg/cluster/spec/instance.go]: Have been deleted

File: pkg/cluster/spec/monitoring.go (hunk 2 above)
Comment (WalterWj) [reply pkg/cluster/spec/monitoring.go]: Have been updated

**Review by srstack — APPROVED** [review]

**Comment by kaaaaaaang** [comment]
I think “additional_args” is a more clear name than "more_command"

**Comment by WalterWj** [comment]
Update run_command to additional_args
image
image

**Comment by WalterWj** [comment]
> I think “additional_args” is a more clear name than "more_command"

Done

**Review by WalterWj — COMMENTED** [review]
lgtm

**Review by srstack — APPROVED** [review]

**Pull request merged**